dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessDefinitionServiceImpl.java

```diff
@@ -300,7 +301,7 @@
             if (taskNodeList.size() != taskRelationList.size()) {
                 Set<Long> postTaskCodes = taskRelationList.stream().map(ProcessTaskRelationLog::getPostTaskCode).collect(Collectors.toSet());
                 Set<Long> taskNodeCodes = taskNodeList.stream().map(TaskNode::getCode).collect(Collectors.toSet());
-                Collection<Long> codes = CollectionUtils.subtract(postTaskCodes, taskNodeCodes);
+                Collection<Long> codes = org.apache.commons.collections4.CollectionUtils.subtract(postTaskCodes, taskNodeCodes);
```

Comment:
Is seems we can only use `CollectionUtils.subtract(postTaskCodes, taskNodeCodes)` here?

Reply:
> Is seems we can only use `CollectionUtils.subtract(postTaskCodes, taskNodeCodes)` here?

Nope, there is another `CollectionUtils` from `org.apache.commons.collections`, not `4`